### PR TITLE
CR-1116717 ASTER TC 18.01 - xbutil dmatest: "ERROR: DMA failed: No such file or directory" when running test for 256MB

### PR DESCRIPTION
**Pull request by chienwei-lan**
#### Problem solved by the commit
The issue is calling xclSyncBo with non-exist BO handle. In this case, we have only one BO and two DMA threads, adjust_e will be b+2 which is not true and the number should be b+1 since we only have one BO, choose the smaller number between BO list size and DMA threads

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
#4971

### CODE REVIEW

**Review by xuhz — APPROVED** [review]

**Comment by chienwei-lan** [comment]
retest this please

File: src/runtime_src/core/pcie/common/dmatest.h [hunk 1]
```diff
@@ -86,7 +86,7 @@ namespace xcldev {
             }
 
             auto len = ((e - b) < count) ? 1 : (e - b)/count;
-            const auto ajust_e = b + len * count;
+            const auto ajust_e = b + len * std::min(count, (unsigned)len);
```
Comment (stsoe) [inline src/runtime_src/core/pcie/common/dmatest.h:89]:
```suggestion
            const auto ajust_e = b + len * std::min<unsigned int>(count, len);
```

**Review by stsoe — CHANGES_REQUESTED** [review]

**Review by stsoe — APPROVED** [review]

**Comment by dayeh-xilinx** [comment]
retest this please. Infra error caused by github.com fetch.

**Comment by sabarinaxil** [comment]
retest this please

**Comment by xilinxgitops** [comment]
retest this please

**Pull request merged**